server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -1714,23 +1720,37 @@ public AccountVO updateAccount(UpdateAccountCmd cmd) {
         // Check if user performing the action is allowed to modify this account
         checkAccess(getCurrentCallingAccount(), _domainMgr.getDomain(account.getDomainId()));
 
-        // check if the given account name is unique in this domain for updating
-        Account duplicateAcccount = _accountDao.findActiveAccount(newAccountName, domainId);
-        if (duplicateAcccount != null && duplicateAcccount.getId() != account.getId()) {
-            throw new InvalidParameterValueException(
-                    "There already exists an account with the name:" + newAccountName + " in the domain:" + domainId + " with existing account id:" + duplicateAcccount.getId());
+        if(newAccountName != null) {
```

Comment:
You can use `StringUtils.isEmpty` here. Then, you do not need all of these netsting

Reply:
@rafaelweingartner 
Hm, I don't think so or don't get the idea. I would like to distinguish three situations:
1. newAccountName is not specified
2. newAccountName is specified but empty string -> exception 1
3. newAccountName is specified and not empty string -> action or exception

So, don't get how `StringUtils.isEmpty()` helps.

Reply:
@rafaelweingartner But how it will behave if `newAccountName` is null which is normal if API is not intending to change that. It is a normal case and whole code block must be skipped.

Reply:
good catch. I was not paying attention to this detail. Then, I would simply do `newAccountName != null && StringUtils.isBlank(newAccountName)`. 

That is only a suggestion. You can leave the code as is if you think it is better this way.

Reply:
@rafaelweingartner ok. I'll leave as is.